### PR TITLE
Adding litecoin and litecoin testnet support

### PR DESCRIPTION
**Pull request by kaykurokawa**
Adding litecoin and litecoin testnet support

### CODE REVIEW

File: altcoin/core/__init__.py [hunk 1]
```diff
@@ -216,6 +216,17 @@ class CoreDogeTestNetParams(CoreDogeMainParams):
     NAME = 'dogecoin_test'
     GENESIS_BLOCK = CAltcoinBlock.deserialize(x('010000000000000000000000000000000000000000000000000000000000000000000000696ad20e2dd4365c7459b4a4a5af743d5e92c6da3229e6532cd605f6533f2a5bb9a7f052f0ff0f1ef7390f000101000000010000000000000000000000000000000000000000000000000000000000000000ffffffff1004ffff001d0104084e696e746f6e646fffffffff010058850c020000004341040184710fa689ad5023690c80f3a49c8f13f8d45b8c857fbcbc8bc4a8e4d3eb4b10f4d4604fa08dce601aaf0f470216fe1b51850b4acf21b179c45070ac7b03a9ac00000000'))
 
+class CoreLtcMainParams(bitcoin.core.CoreChainParams):
```
Comment (kanzure) [inline altcoin/core/__init__.py:219]: Would be nice if these are linked to where in the Litecoin source code they are pulled. (in comments in the source code)

**Comment by rnicoll** [comment]
Looks good by eye, want to test it properly before I merge, may be this weekend, but will try for before!

**Comment by kaykurokawa** [comment]
Hello, I added comment links to the litecoin source code as requested. 
I also corrected proof of work limit on dogecoin /litecoin , they were 2^256-1 >> 32 when it should be 2^256-1 >> 20  

https://github.com/litecoin-project/litecoin/blob/master-0.8/src/main.cpp#L39
https://github.com/dogecoin/dogecoin/blob/master/src/chainparams.cpp#L115

**Comment by rnicoll** [comment]
Good catch with the proof of work limit. Trying to figure out a good test protocol now.

**Pull request merged**